rts/motoko-rts-tests/src/gc/heap.rs

```diff
@@ -204,11 +204,21 @@ impl MotokoHeapInner {
             map.len(),
         );
 
-        let mut heap: Vec<u8> = vec![0; heap_size];
+        let mut heap: Vec<u8> = vec![0; heap_size + 28];
+
+        // MarkCompact assumes that the dynamic heap starts at a 32-byte multiple
+        let realign = match gc {
+            GC::Copying => 0,
+            GC::MarkCompact => (32 - &heap[static_heap_size_bytes] as *const u8 as usize % 32) % 32,
```

Comment:
Something like this should be simpler:

```suggestion
            GC::MarkCompact => (32 - heap.as_ptr() as usize % 32) % 32,
```

Reply:
Oops that should be
``` diff
            GC::MarkCompact => (32 - heap.as_ptr().add(static_heap_size_bytes) as usize % 32) % 32,
```

Reply:
And that seems to work nicely.